collated/main/A0148096W.md

```diff
@@ -0,0 +1,1230 @@
+        String deadline = setToNullIfIsEmptyString(parsed.get(HideCommand.KEYWORD_DEADLINE));
+        String startTime = setToNullIfIsEmptyString(parsed.get(HideCommand.KEYWORD_PERIOD_START_TIME));
+        String endTime = setToNullIfIsEmptyString(parsed.get(HideCommand.KEYWORD_PERIOD_END_TIME));
+        String tags = setToNullIfIsEmptyString(parsed.get(HideCommand.KEYWORD_TAG));
+
+        if(tags == null){
+            tags = "";
+        } else {
+            tags = removeFullStopsAndCommas(tags);
+        }
+
+        try {
+            return new HideCommand(
+                    type,
+                    date,
+                    deadline,
+                    startTime,
+                    endTime,
+                    getTagsFromArgs(tags)
+            );
+        } catch (IllegalValueException ive) {
+            return new IncorrectCommand(ive.getMessage());
+        }
+    }
+
+```
+###### \java\teamfour\tasc\logic\parser\Parser.java
+``` java
+    /**
+     * Parses arguments in the context of the change calendar view command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareCalendar(String args) {
+        if (args.equals("")) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, CalendarCommand.MESSAGE_USAGE));
+        }
+        try {
+            return new CalendarCommand(args.trim());
+        } catch (IllegalValueException ive) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, CalendarCommand.MESSAGE_USAGE));
+        }
+    }
+
+```
+###### \java\teamfour\tasc\logic\parser\Parser.java
+``` java
+    /**
+     * Parses arguments in the context of the undo command.
+     * Special case: if no arg is provided, undoes 1 command.
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareUndo(String args) {
+        if (args.equals("")) {
+            try {
+                return new UndoCommand();
+            } catch (IllegalValueException ive) {
+                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UndoCommand.MESSAGE_USAGE));
+            }
+        }
+        Optional<Integer> index = parseIndex(args);
+        if(!index.isPresent()){
+            return new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, UndoCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            return new UndoCommand(index.get());
+        } catch (IllegalValueException ive) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UndoCommand.MESSAGE_USAGE));
+        }
+    }
+    
+```
+###### \java\teamfour\tasc\model\history\HistoryItem.java
+``` java
+
+package teamfour.tasc.model.history;
+
+/**
+ * An object that can be reverted to past states.
```

Comment:
interface technically cannot be instantiated right? in this context, please reword this comment.